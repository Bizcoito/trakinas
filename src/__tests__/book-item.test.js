import React from 'react';
import BookItem from '../components/book-item';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe('BookItem', () => {
  const book = {
    bookId: '123qwe',
    name: 'Xunda',
    description: 'Damironga',
    thumbnail: 'http://pix.toile-libre.org/upload/original/1499647888.png',
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(<BookItem key={book.bookId} book={book} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders props on html', () => {
    const component = mount(<BookItem key={book.bookId} book={book} />);
    expect(component.find('img.media-object').props().src).toEqual(book.thumbnail);
    expect(component.find('h4').text()).toEqual(book.name);
    expect(component.find('p').text()).toEqual(book.description);
    expect(component.find('BookActionButton').props().book).not.toBeNull();
  });
});
