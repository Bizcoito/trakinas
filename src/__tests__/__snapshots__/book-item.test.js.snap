// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookItem renders correctly 1`] = `
<div
  className="list-group-item media row"
>
  <div
    className="media-left book-thumbnail"
  >
    <img
      className="media-object"
      src="http://pix.toile-libre.org/upload/original/1499647888.png"
    />
  </div>
  <div
    className="media-body"
  >
    <div
      className="media-heading"
    >
      <h4>
        Xunda
      </h4>
      <p>
        Damironga
      </p>
      <div>
        <button
          className="btn btn-info"
          onClick={[Function]}
        >
          Return book
        </button>
      </div>
    </div>
  </div>
</div>
`;
